exercises/alphametics/test/Tests.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Data.List (sort)
+
+import Alphametics (solve)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "alphametics" $
+          describe "solve" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion  = solution `shouldBe` sort <$> expected
+        solution = sort <$> solve puzzle
+
+-- Test cases adapted from `exercism/x-common/exercises/alphametics/acanonical-data.json` on 2016-10-26.
+
+data Case = Case { description  ::        String
+                 , puzzle       ::        String
+                 , expected     ::        Maybe[(Char, Int)]
+                 }
```

Comment:
##### Cosmetic suggestion

A lot of unneeded spaces. One missing space after `Maybe`.

I think that this would be more readable:
```haskell
data Case = Case { description :: String
                 , puzzle      :: String
                 , expected    :: Maybe [(Char, Int)]
                 }
```